### PR TITLE
Add type validation for additionalProperties

### PR DESCRIPTION
**Pull request by ankhyk**

### CODE REVIEW

File: regression/__snapshots__/validateExamplesRegression_latestOnly.test.ts.snap [hunk 1]
```diff
@@ -5653,6 +5653,462 @@ Array [
...
   },
+  Object {
+    "code": "INVALID_TYPE",
+    "exampleJsonPath": "$responses.200.body.value[0].properties.activities[0].typeProperties.activities[0]['inputs']",
+    "examplePosition": Object {
+      "column": 35,
+      "line": 46,
+    },
+    "exampleUrl": "/home/vsts/work/1/s/regression/azure-rest-api-specs/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/Pipelines_ListByFactory.json",
+    "message": "Expected type object but found type array",
```
Comment (raych1) [inline regression/__snapshots__/validateExamplesRegression_latestOnly.test.ts.snap]: Can you check if these new errors are expected and caused by this PR change?

Comment (ankhyk) [reply regression/__snapshots__/validateExamplesRegression_latestOnly.test.ts.snap]: some errors are unexpected, I'll update the rules

File: lib/swagger/swaggerTypes.ts [hunk 2]
```diff
@@ -268,6 +268,9 @@ export interface Schema extends BaseSchema {
 
   // x-ms-discriminator-value exists but discriminator missing
   _missingDiscriminator?: boolean;
+
+  // Additional property support
+  additionalObjectTypeProperties?: boolean;
```
Comment (raych1) [inline lib/swagger/swaggerTypes.ts]: Can you rename it to `additionalPropertiesWithObjectType`

File: regression/__snapshots__/validateExamplesRegression_latestOnly.test.ts.snap (hunk 1 above)
Comment (raych1) [reply regression/__snapshots__/validateExamplesRegression_latestOnly.test.ts.snap]: Did you make the change? I don't see updates to this PR.

**Comment by ankhyk** [comment]
I validated the five snapshot test and the result was as expected

**Review by raych1 — APPROVED** [review]

**Pull request merged**